src/array.c

```diff
@@ -235,6 +234,32 @@ JL_DLLEXPORT jl_array_t *jl_reshape_array(jl_value_t *atype, jl_array_t *data,
     return a;
 }
 
+JL_DLLEXPORT jl_array_t *jl_string_to_array(jl_value_t *str)
+{
+    jl_ptls_t ptls = jl_get_ptls_states();
+    jl_array_t *a;
+
+    int ndimwords = jl_array_ndimwords(1);
+    int tsz = JL_ARRAY_ALIGN(sizeof(jl_array_t) + ndimwords*sizeof(size_t) + sizeof(void*), JL_SMALL_BYTE_ALIGNMENT);
+    a = (jl_array_t*)jl_gc_alloc(ptls, tsz, jl_array_uint8_type);
+    a->flags.pooled = tsz <= GC_MAX_SZCLASS;
+    a->flags.ndims = 1;
+    a->offset = 0;
+    a->data = jl_string_data(str);
+    a->flags.isaligned = 0;
+    a->elsize = 1;
+    a->flags.ptrarray = 0;
+    jl_array_data_owner(a) = str;
+    a->flags.how = 3;
+    a->flags.isshared = 0;
```

Comment:
Yes; `resize!` usually gets called on an IOBuffer's storage before `take!`, and if `isshared` is set it causes the data to be copied and we lose performance.